projects/observability/src/shared/dashboard/data/graphql/table/entity/mode-entity-table-data-source.model.ts

```diff
@@ -0,0 +1,65 @@
+import { assertUnreachable } from '@hypertrace/common';
+import { TableDataRequest, TableDataResponse, TableMode, TableRow } from '@hypertrace/components';
+import { ModelTemplatePropertyType } from '@hypertrace/dashboards';
+import {
+  GraphQlFilter,
+  SpecificationBackedTableColumnDef,
+  TableDataSourceModel
+} from '@hypertrace/distributed-tracing';
+import { Model, ModelJson, ModelProperty } from '@hypertrace/hyperdash';
+
+@Model({
+  type: 'mode-entity-table-data-source'
```

Comment:
I was actually imagining this type of switching model a level up, switching between different table widget models based on mode that defined different sets of columns (and potentially different data sources, too). It'd still need a way of passing mode down from the renderer to the widget model, but it looks like that's already a requirement anyway.

Reply:
A more concrete pseudo-json:
```typescript
{
    type: 'mode-toggle-table-widget',
    flat: {
       type: 'table-widget',
       style: TableStyle.FullPage,
       searchAttribute: 'name',
       columns: [
          {
              type: 'table-widget-column',
              title: 'Name',
              width: '30%',
              display: ObservabilityTableCellType.Entity,
              value: {
                type: 'entity-specification',
                'entity-type': ObservabilityEntityType.Api
              }
         }
       ],
       data: {
        type: 'entity-table-data-source',
        entity: 'API'
       }
    },
    tree: {
       ....
    }
  
}
```

The benefit I see in this approach is that the column models and table component can remain as is, with only a new model being added (and likely a couple changes in renderer to communicate the mode). It also leaves flexibility to use different search attributes based on mode, different data sources, change column orders, etc. - basically, it's two completely different tables, which is really how it is in the UI anyway.

Reply:
Well I finally found the example I wrote up before and... I clearly led down the wrong path, sorry about that. https://github.com/hypertrace/hypertrace-ui/pull/321#discussion_r516949368
The goal is the same here, but because the columns aren't in the data source (my mistake there) the switch moves up a level.